Bottom: 548bf42a3e4cfc813e69f1f97570a4ce8f972360
Top:    003037aaed9bc101519e6d2fd763fa0409f2efc6
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-07-12 11:12:22 -0500

Populate buildbot with gitlab public keys


---
diff --git a/ansible-playbook/buildbot.yml b/ansible-playbook/buildbot.yml
index e14b93b9..bc6c7663 100644
--- a/ansible-playbook/buildbot.yml
+++ b/ansible-playbook/buildbot.yml
@@ -7,12 +7,17 @@
   become_method: sudo
   tags:
   - buildbot
+  pre_tasks:
+  - name: Scan Gitlab ssh keys
+    command: ssh-keyscan {{gitlab_hostname}}
+    register: gitlab_keys
   roles:
   - name: Build and start the Buildbot master container
     role: buildbot-master
     tags:
     - buildbot-master
-
+    vars:
+      known_hosts: "{{gitlab_keys.stdout}}"
   - name: Add buildbot user to Gitlab
     role: gitlab-user
     vars:
diff --git a/docker/buildbot-master/buildbot/start.sh b/docker/buildbot-master/buildbot/start.sh
index 85c28248..7b5acf05 100755
--- a/docker/buildbot-master/buildbot/start.sh
+++ b/docker/buildbot-master/buildbot/start.sh
@@ -8,6 +8,12 @@ BUILDBOT_DATA=${BUILDBOT_DATA-"/buildbot"}
 
 ssh-keygen -q -b ${SSH_BITS} -t ${SSH_TYPE} -N '' -f ${SSH_FILE} 0>&- 1>/dev/null 2>/dev/null
 
+if [ -n "$BUILDBOT_KNOWN_HOSTS_FILE" ] && [ -f "$BUILDBOT_KNOWN_HOSTS_FILE" ]
+then
+    mkdir -p "${HOME}/.ssh"
+    cp -v "$BUILDBOT_KNOWN_HOSTS_FILE" "${HOME}/.ssh/known_hosts"
+fi
+
 ln -v -s -f ${HOME}/buildbot.tac ${HOME}/master.cfg ${HOME}/*.py $BUILDBOT_DATA
 
 # Run original buildbot entrypoint
diff --git a/roles/buildbot-master/defaults/main.yml b/roles/buildbot-master/defaults/main.yml
index 9bc9ac3c..cc654992 100644
--- a/roles/buildbot-master/defaults/main.yml
+++ b/roles/buildbot-master/defaults/main.yml
@@ -33,6 +33,7 @@ config_hostdir: '{{ docker_restore_config_base_dir }}/{{ container_name }}'
 # environment passed to the container
 container_env:
   BUILDBOT_DATABASE: 'sqlite:///{{ data_path }}/state.sqlite'
+  BUILDBOT_KNOWN_HOSTS_FILE: '{{secrets_path}}/{{secrets_known_hosts}}'
   BUILDBOT_SMTP_HOST: 'relay.dnsexit.com'
   BUILDBOT_SMTP_USER: 'Dean.cox'
   BUILDBOT_SMTP_PASS: 'no5156'
@@ -60,7 +61,7 @@ container_networks:
 # volumes mounted within the container
 container_volumes:
   - '{{ data_volume }}:{{ data_path }}:z'
-  - '{{ config_hostdir }}:{{ secrets_path }}'
+  - '{{ secrets_hostdir }}/{{secrets_known_hosts}}:{{ secrets_path }}/{{secrets_known_hosts}}'
 
 # mount path of the data volume
 data_path: '/buildbot'
diff --git a/roles/buildbot-master/tasks/main.yml b/roles/buildbot-master/tasks/main.yml
index d7d69e36..8d1a8400 100644
--- a/roles/buildbot-master/tasks/main.yml
+++ b/roles/buildbot-master/tasks/main.yml
@@ -10,6 +10,16 @@
   docker_volume:
     name: '{{ data_volume }}'
 
+- name: Create secrets volume
+  file:
+    path: '{{ secrets_hostdir }}'
+    state: directory
+
+- name: Populate known_hosts secret
+  copy:
+    content: '{{ known_hosts }}'
+    dest: '{{ secrets_hostdir }}/{{ secrets_known_hosts }}'
+
 - name: Checkout image repo
   git:
     repo: '{{ image_repo }}'
